### ISSUE TITLE
Authentication degraded v4.2

### ISSUE THREAD

**Issue by oswee** [openshift/cluster-authentication-operator#209]
Can't get it to work anymore. Cluster setup was working fine for few times, but now every time i get Authentication degraded and so console does not come up as well. Everything else seems ok.
This is what is see in logs:

```sh
~ oc logs authentication-operator-59bd6dffb8-xhssv -n openshift-authentication-operator

I1024 13:05:25.642299       1 status_controller.go:165] clusteroperator/authentication diff {"status":{"conditions":[{"lastTransitionTime":"2019-10-24T12:52:00Z","message":"RouteHealthDegraded: failed to GET route: EOF","reason":"RouteHealthDegradedFailedGet","status":"True","type":"Degraded"},{"lastTransitionTime":"2
019-10-24T12:47:26Z","reason":"NoData","status":"Unknown","type":"Progressing"},{"lastTransitionTime":"2019-10-24T12:47:26Z","reason":"NoData","status":"Unknown","type":"Available"},{"lastTransitionTime":"2019-10-24T12:47:26Z","reason":"AsExpected","status":"True","type":"Upgradeable"}]}}
I1024 13:05:25.668003       1 event.go:209] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-authentication-operator", Name:"authentication-operator", UID:"af3b5e80-f65b-11e9-abe5-001a4a160128", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'OperatorStatusChanged' Sta
tus for clusteroperator/authentication changed: Degraded message changed from "ResourceSyncControllerDegraded: Post https://172.30.0.1:443/api/v1/namespaces/openshift-config-managed/configmaps: http2: server sent GOAWAY and closed the connection; LastStreamID=479, ErrCode=NO_ERROR, debug=\"\"\nRouteHealthDegraded: fai
led to GET route: EOF" to "RouteHealthDegraded: failed to GET route: EOF"
```

```sh
~ oc version
Client Version: openshift-clients-4.2.0-201910041700
Server Version: 4.2.0
Kubernetes Version: v1.14.6+2e5ed54
```
I have DNS and HAProxy in front of it which was working well before.

What i did so far:
1) Downloaded latest installer from given link @ [Install on Bare Metal: User-Provisioned Infrastructure](https://cloud.redhat.com/openshift/install/metal/user-provisioned)
2) Downloaded given pull_secred
3) Created install-config.yaml with given pull_secret and ssh key
4) Generated new ignition configs.
5) Uploaded those to Matchbox and restarted service
6) Downloaded given RHCOS images and uploaded to TFTP
7) Did iPXE boot

Basically ensured that i have all latest versions of all required components.
